astropy/io/misc/parquet.py

```diff
@@ -0,0 +1,307 @@
+                if schema_only:
+                    # Choose an arbitrary string length since
+                    # are not reading in the table.
+                    strlen = 10
+                    warnings.warn(f"No table::len::{col} found in metadata. "
+                                  f"Guessing {{strlen}} for schema.",
+                                  AstropyUserWarning)
+                else:
+                    strlen = max([len(row.as_py()) for row in pa_table[col]])
+                    warnings.warn(f"No table::len::{col} found in metadata. "
+                                  f"Using longest string ({{strlen}} characters).",
+                                  AstropyUserWarning)
+            if schema.field(col).type == pa.string():
+                dtype.append(f'U{strlen}')
+            else:
+                dtype.append(f'|S{strlen}')
+        else:
+            dtype.append(schema.field(col).type.to_pandas_dtype())
+
+    data = np.zeros(num_rows, dtype=list(zip(columns_to_read, dtype)))
+
+    if not schema_only:
+        for name in columns_to_read:
+            data[name][:] = pa_table[name].to_numpy()
+
+    table = Table(data=data)
+    if meta_dict != {}:
+        table.meta = meta_dict
+
+    if meta_hdr is not None:
+        header_cols = dict((x['name'], x) for x in meta_hdr['datatype'])
+        for col in table.columns.values():
+            for attr in ('description', 'format', 'unit', 'meta'):
+                if attr in header_cols[col.name]:
+                    setattr(col, attr, header_cols[col.name][attr])
+
+    table = serialize._construct_mixins_from_columns(table)
+
+    return table
+
+
+def write_table_parquet(table, output, overwrite=False):
+    """
+    Write a Table object to a Parquet file
+
+    This requires `pyarrow <https://arrow.apache.org/docs/python/>`_
+    to be installed.
+
+    Parameters
+    ----------
+    table : `~astropy.table.Table`
+        Data table that is to be written to file.
+    output : str
+        The filename to write the table to.
+    overwrite : bool
+        Whether to overwrite any existing file without warning.
+    """
+
+    from astropy.table import meta, serialize
+    from astropy.utils.data_info import serialize_context_as
+    try:
+        import pyarrow as pa
+        from pyarrow import parquet
+    except ImportError:
+        raise Exception("pyarrow is required to read and write Parquet files")
+
+    if not isinstance(output, str):
+        raise TypeError('output should be a string')
+
+    if os.path.exists(output):
+        if overwrite:
+            os.remove(output)
+        else:
+            raise OSError(f"File exists: {output}")
+
+    with serialize_context_as('parquet'):
+        encode_table = serialize.represent_mixins_as_columns(table)
+    meta_yaml = meta.get_yaml_from_table(encode_table)
+    meta_yaml_str = '\n'.join(meta_yaml)
+
+    metadata = {}
+    for col in encode_table.columns:
```

Comment:
I'd suggest `for name, col in encode_table.columns.items():`. In the table code the preferred convention is for `col` to refer to a column object, not the name of a column, while `name` is the string name. In this case doing it like this will save a lot of repetition with the `encode_table[col]`.